nix/flake-module.nix

```diff
@@ -63,6 +63,20 @@ in
                   Build tools useful for Haskell development are included by default.
                 '';
               };
+              extraLibraries = mkOption {
+                type = functionTo (types.attrsOf (types.nullOr types.package));
```

Comment:
When would someone want to use `null` as a value?

Reply:
`null` just happens to be usable in places where a package is expected. This is somewhat useful; for example `base` is defined as `null` because it's implicitly included everywhere already.
Using `null` instead of, say, `lib.optionalAttrs` to remove the entire attribute makes some code simpler and lazier and therefore more recursion friendly: if `a.c` is nullable, `a.b` can be evaluated by evaluating subexpression `a`'s attrset, without `c`'s condition, whereas if `c`'s presence is optional, the condition has to be computed as part of subexpression `a` in `a.b`.

Reply:
I was wondering why would someone want to use `null` as a value here in `extraLibraries` in particular (as an actual use case) - but it sounds like you are saying this type is generally useful in any context; okay.

Reply:
Yeah, I don't think anyone will set `base` or `text` or `Cabal` here, but if they do, it should be accepted, as Nixpkgs Haskell does.